l1-contracts/deploy-scripts/PrepareZKChainRegistrationCalldata.s.sol

```diff
@@ -0,0 +1,286 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+// solhint-disable no-console
+
+import {Script, console2 as console} from "forge-std/Script.sol";
+import {stdToml} from "forge-std/StdToml.sol";
+
+import {IBridgehub} from "contracts/bridgehub/IBridgehub.sol";
+import {Bridgehub} from "contracts/bridgehub/Bridgehub.sol";
+import {L2ContractHelper} from "contracts/common/libraries/L2ContractHelper.sol";
+import {AddressAliasHelper} from "contracts/vendor/AddressAliasHelper.sol";
+import {L1SharedBridge} from "contracts/bridge/L1SharedBridge.sol";
+import {IGovernance} from "contracts/governance/IGovernance.sol";
+import {Utils} from "./Utils.sol";
+
+contract PrepareZKChainRegistrationCalldataScript is Script {
+    using stdToml for string;
+
+    address internal constant ADDRESS_ONE = 0x0000000000000000000000000000000000000001;
+
+    struct Config {
+        address chainAdmin;
+        address bridgehub;
+        address stateTransitionProxy;
```

Comment:
This is STM address, right?

it already has the mapping to bridgehub - so maybe we don't need to pass bridgehub at all?
(the more things we can 'auto-detect' the less errors people will do).

Reply:
seems that bridgehub is still in the config..

Reply:
it's in the struct so we don't bother getting it every time from the shared_bridge, but it's doesn't have to be provided by the user of the script https://github.com/matter-labs/era-contracts/pull/671/files#diff-342e319e208c78c91b276c94c5dd88420f11857dcb5ebcee28c5376f0bb60bf1R135

Reply:
ah, so the parts of config are coming from 'user' and parts are computed..

In such case, I'd suggest clearly separating them in the file here -- so that we put (for example) user stuff at the top of this config, and 'fetched/computed' stuff later.
(preferably I'd have 2 separate structs - but this is up to you).